drivers/include/nrf24l01p.h

```diff
@@ -1,627 +1,1366 @@
+    gpio_t pin_cs;      /**< SPI chip select gpio pin */
+    gpio_t pin_ce;      /**< NRF24L01P chip enable gpio pin */
+    gpio_t pin_irq;     /**< NRF24L01P IRQ gpio pin */
+    union {
+        struct {
+            /**
+             * @brief   Array to access the addresses of pipe 0 and pipe 1 via
+             *          pipe indices @see nrf24l01p_pipe_t
+             */
+            uint8_t rx_addr_long[2][NRF24L01P_MAX_ADDR_WIDTH];
+            /**
+             * @brief   Array to access the addresses of pipe 2, pipe 3,
+             *          pipe 4 and pipe 5 via pipe indices
+             *          @see nrf24l01p_pipe_t
+             */
+            uint8_t rx_addr_short[4];
+        } arxaddr; /**< Rx addresses as arrays */
+        struct {
+            /**
+             * @brief   pipe 0 Rx address
+             */
+            uint8_t rx_pipe_0_addr[NRF24L01P_MAX_ADDR_WIDTH];
+            /**
+             * @brief   pipe 1 Rx address
+             */
+            uint8_t rx_pipe_1_addr[NRF24L01P_MAX_ADDR_WIDTH];
+            /**
+             * @brief   pipe 2 Rx address
+             */
+            uint8_t rx_pipe_2_addr;
+            /**
+             * @brief   pipe 3 Rx address
+             */
+            uint8_t rx_pipe_3_addr;
+            /**
+             * @brief   pipe 4 Rx address
+             */
+            uint8_t rx_pipe_4_addr;
+            /**
+             * @brief   pipe 5 Rx address
+             */
+            uint8_t rx_pipe_5_addr;
+        } rxaddrpx; /**< Rx addresses as named variables */
+    } urxaddr; /**< Union of Rx addresses as named variables and as arrays */
+    nrf24l01p_cfg_t config; /**< Current configuration values */
+} nrf24l01p_params_t;
+
+/**
+ * @brief   NRF24L01P device struct
+ */
+typedef struct {
+    netdev_t netdev;            /**< Netdev member */
+    nrf24l01p_params_t params;  /**< Parameters */
+    /**
+     * @brief Destination address as PTX
+     *        A PTX node must change pipe 0 Rx address to Tx address in order to receive ACKs.
+     *        If node switches back to Rx mode, pipe 0 Rx address must be restored from params.
+     */
+    uint8_t tx_addr[NRF24L01P_MAX_ADDR_WIDTH];
+    uint8_t tx_addr_len;        /**< Tx address length */
+    uint8_t state;              /**< Current operation state */
+#if !defined (NDEBUG) || defined(DOXYGEN)
+    uint8_t have_spi_access;    /**< != 0: dev already has SPI bus acquired*/
+    uint8_t transitions;        /**< Possible transitions from current state */
+#endif
+} nrf24l01p_t;
+
+/**
+ * @brief Get state variable as a string
+ *
+ * @param[in] state     State
+ *
+ * @return              @p state as a string
+ */
+const char *nrf24l01p_state_to_string(nrf24l01p_state_t state);
+
+/**
+ * @brief Convert string to state variable
+ *
+ * @param[in] sstate    State string
+ *
+ * @return              State variable
+ */
+nrf24l01p_state_t nrf24l01p_string_to_state(const char *sstate);
```

Comment:
Is this really needed?

Reply:
The datasheet states:

> nRF24L01+ must be in a standby or power down mode before writing to the configuration registers

(page 49)
So e.g. to configure CRC, the device should be in one of those states.
So the device must be put into one of those states somehow.
You can do that with `ifconfig` or `nrf24ctl`
For `nrf24ctl` the command to put device 0 to sleep mode would be:
`nrf24ctl 0 --set state POWER_DOWN`
Here the "POWER_DOWN" string is translated to NRF24L01P_STATE_POWER_DOWN.

That was the reason for the function.

Reply:
OK, I see.